src/python_testing/TC_IDM_4_2.py

```diff
@@ -0,0 +1,589 @@
+
+            idleModeDuration = await self.get_idle_mode_duration(CR1)
+
+            SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT = idleModeDuration
+            logging.info("SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT: " + idleModeDuration + " sec")
+        else:
+            # Defaulting SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT to 60 minutes
+            logging.info("Set SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT = 60 mins")
+            SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT = 3600
+
+        '''
+        ##########
+        Step 1
+        ##########
+        '''
+        self.print_step(1, "CR1 sends a subscription message to the DUT with MaxIntervalCeiling set to a value greater than SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT. DUT sends a report data action to the TH. CR1 sends a success status response to the DUT. DUT sends a Subscribe Response Message to the CR1 to activate the subscription.")
+        min_interval_floor_sec = SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT - 60
+        max_interval_ceiling_sec = SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT + 60
+
+        # Subscribe to attribute
+        sub_cr1_step1 = await CR1.ReadAttribute(
+            nodeid=self.dut_node_id,
+            attributes=node_label_attr_path,
+            reportInterval=(min_interval_floor_sec, max_interval_ceiling_sec),
+            keepSubscriptions=False
+        )
+
+        # Verify attribute data came back
+        self.verify_attribute_data(
+            sub=sub_cr1_step1,
+            cluster=Clusters.BasicInformation,
+            attribute=node_label_attr
+        )
+
+        # Verify subscriptionId is of uint32 type
+        asserts.assert_true(self.is_uint32(sub_cr1_step1.subscriptionId), "subscriptionId is not of uint32 type.")
+
+        # Verify MaxInterval is of uint32 type
+        sub_cr1_step1_intervals = sub_cr1_step1.GetReportingIntervalsSeconds()
+        sub_cr1_step1_min_interval_floor_sec, sub_cr1_step1_max_interval_ceiling_sec = sub_cr1_step1_intervals
+        asserts.assert_true(self.is_uint32(sub_cr1_step1_max_interval_ceiling_sec), "MaxInterval is not of uint32 type.")
+
+        # Verify MaxInterval is less than or equal to MaxIntervalCeiling
+        asserts.assert_true(sub_cr1_step1_max_interval_ceiling_sec <= max_interval_ceiling_sec,
+                            "MaxInterval is not less than or equal to MaxIntervalCeiling")
+
+        sub_cr1_step1.Shutdown()
+
+        '''
+        ##########
+        Step 2
+        ##########
+        '''
+        self.print_step(2, "CR1 sends a subscription message to the DUT with MaxIntervalCeiling set to a value less than SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT. DUT sends a report data action to the CR1. CR1 sends a success status response to the DUT. DUT sends a Subscribe Response Message to the CR1 to activate the subscription.")
+        min_interval_floor_sec = SUBSCRIPTION_MAX_INTERVAL_PUBLISHER_LIMIT - 60
```

Comment:
same comments on limits apply: either assert that max interval is >= 60 or add range settings (and in that case you have to validate that max_interval > min_interval I assue)

Reply:
Updated